## Read the contents of puzzles.csv both in Python and R and send the output of the terminal to Łukasz Kraiński

In [2]:
import pandas as pd

In [3]:
puzzles = pd.read_csv (r'C:\Users\Sara\Downloads\puzzles.csv')
print (puzzles)

         00008 r6k/pp2r2p/4Rp1Q/3p4/8/1N1P2R1/PqP2bPP/7K b - - 0 24  \
0        0000D  5rk1/1p3ppp/pq3b2/8/8/1P1Q1N2/P4PPP/3R2K1 w - ...     
1        0009B  r2qr1k1/b1p2ppp/pp4n1/P1P1p3/4P1n1/B2P2Pb/3NBP...     
2        000aY  r4rk1/pp3ppp/2n1b3/q1pp2B1/8/P1Q2NP1/1PP1PP1P/...     
3        000hf  r1bqk2r/pp1nbNp1/2p1p2p/8/2BP4/1PN3P1/P3QP1P/3...     
4        000rO          3R4/8/K7/pB2b3/1p6/1P2k3/3p4/8 w - - 4 58     
...        ...                                                ...     
2132983  zzzRN  4r2k/1NR2Q1p/4P1n1/pp1p4/3P4/4q3/PP4PP/6K1 w -...     
2132984  zzzTs  r2qrk2/pb1n1ppQ/1p2p3/2pP2b1/3P4/3B1NN1/PP3PPP...     
2132985  zzzUZ  r2qk2r/pp2ppbp/2n3pn/1B1pP3/3P4/5b1P/PP1N1PP1/...     
2132986  zzzco  5Q2/pp3R1P/1kpp4/4p3/2P1P3/3PP2P/Pr2q3/2K5 w -...     
2132987  zzzhI  r3kb1r/ppp2ppp/2n5/3q3b/3P1B2/5N1P/PPP3P1/RN1Q...     

         f2g3 e6e7 b2b1 b3c1 b1c1 h6c1  1765   74  93   493  \
0                  d3d6 f8d8 d6d8 f6d8  1525   74  97  9211   
1                  b6

## for R:

puzzles <- read.csv("C:\\Users\\Sara\\Downloads\\puzzles.csv", header=FALSE)   
head(puzzles)

  V1                                                                   V2                            V3   V4 V5 V6   V7   
1 00008                 r6k/pp2r2p/4Rp1Q/3p4/8/1N1P2R1/PqP2bPP/7K b - - 0 24 f2g3 e6e7 b2b1 b3c1 b1c1 h6c1 1765 74 93  493   
2 0000D                 5rk1/1p3ppp/pq3b2/8/8/1P1Q1N2/P4PPP/3R2K1 w - - 2 27           d3d6 f8d8 d6d8 f6d8 1525 74 97 9211   
3 0009B r2qr1k1/b1p2ppp/pp4n1/P1P1p3/4P1n1/B2P2Pb/3NBP1P/RN1QR1K1 b - - 1 16           b6c5 e2g4 h3g4 d1g4 1102 75 85  503   
4 000aY       r4rk1/pp3ppp/2n1b3/q1pp2B1/8/P1Q2NP1/1PP1PP1P/2KR3R w - - 0 15           g5e7 a5c3 b2c3 c6e7 1320 74 92  395   
5 000hf      r1bqk2r/pp1nbNp1/2p1p2p/8/2BP4/1PN3P1/P3QP1P/3R1RK1 b kq - 0 19           e8f7 e2e6 f7f8 e6f7 1560 76 88  441   
6 000rO                            3R4/8/K7/pB2b3/1p6/1P2k3/3p4/8 w - - 4 58           a6a5 e5c7 a5b4 c7d8 1039 80 85   54   
                                     V8                                    V9   
1 crushing hangingPiece long middlegame https://lichess.org/787zsVup/black#48   
2               advantage endgame short       https://lichess.org/F8M8OS71#53   
3            advantage middlegame short https://lichess.org/4MWQCxQ6/black#32   
4     advantage master middlegame short       https://lichess.org/iihZGl6t#29   
5         mate mateIn2 middlegame short https://lichess.org/71ygsFeE/black#38   
6           crushing endgame fork short      https://lichess.org/tzeeBEc2#115   